crates/containerd-shim-wasm/src/sandbox/instance.rs

```diff
@@ -142,6 +157,170 @@ pub trait Instance {
     fn wait(&self, waiter: &Wait) -> Result<(), Error>;
 }
 
+/// YoukiInstance is a trait that gets implemented by a WASI runtime that
+/// uses youki's libcontainer library as the container runtime.
+/// It provides default implementations for some of the Instance trait methods.
+/// The implementor of this trait is expected to implement the
+///     new_youki()
+///     get_exit_code()
+///     get_id()
+///     get_root_dir()
+///     build_container()
+/// methods.
+#[cfg(feature = "libcontainer")]
+pub trait YoukiInstance {
```

Comment:
thoughts on putting this (and related imports) into its own file?  Then you only need the one `cfg(feature=libcontainer)` and makes it easier to navigate and find all the things.

Reply:
Done.